### ISSUE TITLE
UnhandledTransitionException with voice

### ISSUE THREAD

**Issue by Shadorc** [Discord4J/Discord4J#536]
**To Reproduce:** 
```java
DiscordClient#getChannelById(Snowflake)
    .cast(VoiceChannel.class)
    .flatMap(voiceChannel -> voiceChannel.join(spec -> {}))
    .doOnNext(VoiceConnection::disconnect)
    .block();
```

**Expected Behavior:** The bot joins then leaves the voice channel.

**Actual Behavior:** The bot joins the voice channel then throws an `UnhandledTransitionException` and stay in the voice channel. Stacktrace: https://pastebin.com/8TLj5nFu

**Version:** Discord4J 3.0.6

**Comment by darichey** [comment]
Please include the full stacktrace.

**Comment by Shadorc** [comment]
Stacktrace included.

**Comment by Shadorc** [comment]
I can confirm that this issue is definitively fixed in the `prototype/voice-for-3.1` branch. I will close it once the 3.1 is released.

**Comment by quanticc** [comment]
Moving milestone to 3.1.0.M1 and closing.
